xarray/core/treenode.py

```diff
@@ -415,11 +415,11 @@ def _post_detach(self: Tree, parent: Tree) -> None:
         """Method call after detaching from `parent`."""
         pass
 
-    def _pre_attach(self: Tree, parent: Tree) -> None:
+    def _pre_attach(self: Tree, parent: Tree, name: str) -> None:
```

Comment:
Should these be deleted if they aren't being used?

Reply:
I do still use `_pre_attach` and `_post_attach`, with the modied signature including `name`.